apps/emqx_management/src/emqx_mgmt_api_listeners.erl

```diff
@@ -240,8 +244,8 @@ crud_listener_by_id_on_node(get, #{bindings := #{id := Id, node := Node}}) ->
         Listener ->
             {200, format(Listener)}
     end;
-crud_listener_by_id_on_node(put, #{bindings := #{id := Id, node := Node, body := Conf}}) ->
-    return_listeners(emqx_mgmt:update_listener(atom(Node), Id, Conf));
+crud_listener_by_id_on_node(put, #{bindings := #{id := Id, node := Node}, body := Conf}) ->
+    return_listeners([emqx_mgmt:update_listener(atom(Node), Id, Conf)]);
```

Comment:
emqx_mgmt:update_listern/3 may crash with `error(Reason)`
we should catch it and return a proper HTTP code

Reply:
`return_listeners/1` filters errors from `update_listener/3` and throw.